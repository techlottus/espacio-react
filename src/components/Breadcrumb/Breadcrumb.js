import React, { useEffect } from "react";

export const Breadcrumb = React.memo(({ data, onItem, onBack }) => {
  const breadcrumbRef = React.createRef();

  useEffect(() => {
    breadcrumbRef.current.data = {
      textItems: data.textItems || [],
      textColor: data.textColor || "brand",
      // iconColor: data.iconColor || '',
      // itemFocusColor: data.itemFocusColor || '',
      icon: data.icon || "",
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    breadcrumbRef.current.addEventListener("onItem", onItem);
    breadcrumbRef.current.addEventListener("onBack", onBack);
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-breadcrumb ref={breadcrumbRef}></lottus-breadcrumb>
    </>
  );
});
