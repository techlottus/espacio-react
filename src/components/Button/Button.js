

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


