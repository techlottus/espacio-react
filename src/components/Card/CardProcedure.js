import React, { useEffect } from "react";


import "./CardProcedure.scss";

export const CardProcedure = ({ data, text, price, onClick }) => {

  const cardRef = React.createRef();

  useEffect(() => {
    cardRef.current.data = {
      icon: data.icon || "",
      title: data.title || "",
      text: data.text || "",
      price: data.price || 0,
      color: data.color || "",
      disabled: data.disabled,
      ticket: data.ticket || "",
      date: data.date || "",
      isContent: data.isContent,
    };
  }, [data]); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    cardRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-card-icon style={{ width: "100%", height: "100%" }} ref={cardRef}>
        <div content="" className="container-content">
          <div className="text">
            <p>{text}</p>
          </div>
          <div className="price">
            <p>{price}</p>
          </div>
        </div>
      </lottus-card-icon>
    </>
  );
};
