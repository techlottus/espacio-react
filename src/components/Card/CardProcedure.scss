@import '../../App.scss';

.container-content {
  font-family: "Nunito", sans-serif;

  .text {
      padding-bottom: pixeltorem(31px);
    p {
      font-size: pixeltorem(14px);
      line-height: pixeltorem(19px);
    }
  }
  .price {
    p {
      font-weight: bold;
      font-size: pixeltorem(16px);
      line-height: pixeltorem(22px);
    }
  }
}
