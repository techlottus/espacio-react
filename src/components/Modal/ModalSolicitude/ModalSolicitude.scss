@import "../../../App.scss";

.modal-solicitude {
  color: map-get($palette-neutral, n00);
  font-family: "Nunito", sans-serif;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 20;
  background-color: rgba(0, 0, 0, 0.6);

  .modalbody {
    position: absolute;
    top: 30%;
    right: 27%;
    width: 40%;
    @media only screen and (max-width: map-get($grid-breakpoints, m)) {
      width: 95%;
      right: pixeltorem(12px);
    }
    .solicitudecontent {
      .title {
        font-weight: bold;
        font-size: pixeltorem(18px);
        line-height: pixeltorem(24px);
        padding-bottom: pixeltorem(16px);
      }
      .text {
        font-size: pixeltorem(12px);
        line-height: pixeltorem(16px);
        padding-bottom: pixeltorem(16px);
        .boldtxt {
          font-weight: bold;
        }
      }
      .btnsolicitude {
        .secondtext {
          font-size: pixeltorem(12px);
          line-height: pixeltorem(16px);
          padding-bottom: pixeltorem(16px);
        }
      }
    }
  }
}
