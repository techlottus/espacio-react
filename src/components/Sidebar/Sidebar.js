import React, { useEffect, useRef, useState } from "react";
import "./Sidebar.scss";
import { useDispatch, useSelector } from "react-redux";
import { typesRequestErrors } from "../../constants/error.constant";
import EmptyState from "../EmptyState/EmptyState";
import { getImageOfAssetsMark } from "../../helpers/getImages";
import { sendInfoTM } from "../../tagging/services/sendInfoTagManager";
import { closeSideBartTag } from "../../tagging/flows/procedures/SidebarTag";
import { useHistory } from "react-router";
import {
  handleSuccessSidebar,
  hanldeCardRequestSidebar,
  hanldeTableRequestSidebar,
  typeSliderbar,
} from "./bodySidebar/bodySidebar";

const Sidebar = ({ typeSlide='',show, onClose, outside }) => {
  const [isCheckRequest, setIsCheckRequest] = useState({
    valid: false,
    table: {},
  });
  const siderContent = useRef();
  const dispatch = useDispatch();
  const [cards, setCards] = useState([]);
  const proceduresStore = useSelector((state) => state.procedures);
  const helpCenterStore = useSelector((state) => state.helpCenterDashboard);
  const [error, setError] = useState({
    isError: false,
    msg: null,
  });
  const { texts } = useSelector((state) => state.texts);
  const history = useHistory();

  useEffect(() => {
    switch(typeSlide) {
      case typeSliderbar.procedures:
        if (proceduresStore.requestInquiryAll !== null) {
          setCards([...proceduresStore.requestInquiryAll]);
        }
        break;
      case typeSliderbar.helpCenter:
        if (helpCenterStore.requestInquiryAll !== null) {
          setCards([...helpCenterStore.requestInquiryAll]);
        }
        break;
      default:
        break;
    }
   
  }, [proceduresStore.requestInquiryAll,helpCenterStore.requestInquiryAll]);

  useEffect(() => {
    if (
      proceduresStore.errors[typesRequestErrors.getRequestInquiryProcedures]
        .isError
    ) {
      setError({
        isError: true,
        msg: proceduresStore.errors[
          typesRequestErrors.getRequestInquiryProcedures
        ].msg,
      });
    }

    if (
      helpCenterStore.errors[typesRequestErrors.getRequestAllTicketsHelp]
        .isError
    ) {
      setError({
        isError: true,
        msg: helpCenterStore.errors[
          typesRequestErrors.getRequestAllTicketsHelp
        ].msg,
      });
    }

    return () => {
      setError({
        isError: false,
        msg: null,
      });
    };
  }, [proceduresStore.errors]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        siderContent.current &&
        !siderContent.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    if (show) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [show, proceduresStore.requestInquiryAll,helpCenterStore.requestInquiryAll]);

  const handleError = () => {
    return (
      <>
        <div className="sidebarbody">
          <EmptyState
            img={getImageOfAssetsMark(texts?.procedures?.images?.emptyError)}
            title={error.msg}
          />
        </div>
      </>
    ); 
  };


  const handleIsError = () => {
    return error.isError
      ? handleError()
      : handleSuccessSidebar(
        texts,
        isCheckRequest,
        dispatch,
        setIsCheckRequest,
        history,
        cards,
        hanldeTableRequestSidebar,
        hanldeCardRequestSidebar
        );
  };

  return show ? (
    <div className="sidebar-container">
      <div className="sidebarcontent" ref={siderContent}>
        <div className="sidebarheader">
          <div className="sidebarheadericon">
            <span
              className="material-icons icon iconleft"
              onClick={() => {
                setIsCheckRequest({
                  valid: false,
                  table: {},
                });
                onClose();
                sendInfoTM(window, closeSideBartTag);
              }}
            >
              close
            </span>
          </div>
        </div>
        {handleIsError()}
      </div>
    </div>
  ) : (
    <></>
  );
};

export default Sidebar;
