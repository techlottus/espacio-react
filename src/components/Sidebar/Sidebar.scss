@import "../../App.scss";

.sidebar-container {
  background-color: map-get($palette-overlay, b-50);
  position: fixed;
  right: 0;
  top: 0;
  z-index: 100;
  height: 100vh;
  width: 100%;
  left: 0;
  display: block;

  .sidebarcontent {
    width: 100%;
    height: 100vh;
    background-color: map-get($palette-neutral, n00);
    position: absolute;
    right: 0;
    transition: transform 0.5s ease-in-out;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      width: 35%;
    }

    .sidebarheader {
      @extend .sidebar-header;
    }

    .sidebarbody {
      @extend .sidebar-body;
    }
  }
}

.sidebar-header {
  width: 100%;
  display: flex;
  flex-direction: column;

  .sidebarheadericon {
    width: 100%;
    margin-top: pixeltorem(20px);
    margin-left: pixeltorem(18px);
    margin-bottom: pixeltorem(32px);

    > .iconleft {
      cursor: pointer;
      font-size: pixeltorem(20px);
    }

  }

}

.sidebar-body {
  width: calc(100% - #{pixeltorem(48px)});
  height: calc(100% - #{pixeltorem(76px)});
  padding: 0 pixeltorem(24px);
  overflow-x: hidden;
  overflow-y: auto;
  scrollbar-width: thin;

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    width: calc(100% - #{pixeltorem(96px)});
    padding: 0 pixeltorem(48px);
  }

  .sidebarbodytitle {
    font-size: pixeltorem(24px);
    font-family: "Nunito Bold", sans-serif;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      font-size: pixeltorem(32px);
      text-align: center;
    }
  }

  .sidebarbodyimg {
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;

    img {
      margin-top: pixeltorem(24px);
      width: pixeltorem(120px);
      margin-bottom: pixeltorem(32px);
    }
  }

  .sidebarbodyitems {
    width: 100%;
  }

  .sidebarbodycardrequests {
    width: 100%;

    .sidebarbodycardrequest {
      margin-bottom: pixeltorem(16px);
    }
  }

  .sidebarbodytable {
    margin-top: pixeltorem(24px);

    .iconback {
      display: flex;
      align-items: center;
      margin-bottom: pixeltorem(72px);
      color: palette-colors($brand, p00);
      cursor: pointer;

      span {

        &:first-child {
          margin-right: pixeltorem(12px);
        }

      }
    }
  }
}
