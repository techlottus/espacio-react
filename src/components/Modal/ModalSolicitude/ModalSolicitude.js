import React, { useEffect, useRef, useState } from "react";
import "./ModalSolicitude.scss";
import { Feedback } from "../../Feedback/Feedback";
import {
  btnSolicitude,
  btnSolicitudeError,
  feedbackPaySolicitude,
  feedbackPaySolicitudeError,
} from "../../../constants/Procedures.constant";
import { Button } from "../../Button/Button";
import { useSelector } from "react-redux";

export const ModalSolicitude = ({
  isShow,
  onClose,
  onNext,
  noTicket,
  isError,
  isShowButton,
  outside
}) => {
  const { ticket } = noTicket;
  const [isModal, setIsModal] = useState(false);
  const { texts } = useSelector((state) => state.texts);

  const modalRef = useRef();

  useEffect(() => {
    setIsModal(isShow);
  }, [isShow]);

  useEffect(() => {
    if (isShow) {
      document.body.style.overflowY = "hidden";
    } else {
      document.body.style.overflowY = "auto";
    }
  }, [isShow]);

  useEffect(() => {
    const handleClickOutside = (event) => {
      if (
        modalRef.current &&
        !modalRef.current.contains(event.target) &&
        window.innerWidth > 991
      ) {
        outside();
      }
    };
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const solicitudeContent = () => {
    return (
      <div className="solicitudecontent">
        <h2 className="title"> {texts?.procedures?.modalSolicitude?.title}</h2>
        <p className="text">
          {texts?.procedures?.modalSolicitude?.textTicket1} <span className="boldtxt">{texts?.procedures?.modalSolicitude?.textTicket2} {ticket}.</span> {texts?.procedures?.modalSolicitude?.textTicket3}
        </p>
        {isShowButton ? (
          <div className="btnsolicitude">
            <p className="secondtext">
              {texts?.procedures?.modalSolicitude?.secondText}
            </p>
            <Button
              data={{
                ...btnSolicitude,
                isExpand: window.innerWidth < 991,
              }}
              onClick={() => {
                onNext();
              }}
            />
          </div>
        ) : (
          ""
        )}
      </div>
    );
  };

  const solicitudeContentError = () => {
    return (
      <div className="solicitudecontent">
        <h2 className="title">
          {texts?.procedures?.modalSolicitude?.errorMsg}
        </h2>
        <p className="text">{isError.message}</p>
        <div className="btnsolicitude">
          <Button
            data={{
              ...btnSolicitudeError,
              isExpand: window.innerWidth < 991,
            }}
            onClick={() => {
              onClose();
            }}
          />
        </div>
      </div>
    );
  };

  const handleModal = () => {
    return (
      <div className="modal-solicitude">
        <div className="modalbody" ref={modalRef}>
          <Feedback
            data={
              isError.error ? feedbackPaySolicitudeError : feedbackPaySolicitude
            }
            html={isError.error ? solicitudeContentError : solicitudeContent}
            onRight={() => {
              onClose();
            }}
          />
        </div>
      </div>
    );
  };

  return <>{isModal ? handleModal() : ""}</>;
};
