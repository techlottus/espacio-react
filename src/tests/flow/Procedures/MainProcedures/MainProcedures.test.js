import React from "react";
import { mount } from "enzyme";
import MainProcedures from "../../../../flow/Procedures/components/Main/MainProcedures";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { act } from "react-dom/test-utils";
import { ProceduresContext } from "../../../../flow/Procedures/ProceduresContext";
import { BrowserRouter } from "react-router-dom";
import { ModalSolicitude } from "../../../../components/Modal/ModalSolicitude/ModalSolicitude";
import { CardProcedure } from "../../../../components/Card/CardProcedure";
import Sidebar from "../../../../components/Sidebar/Sidebar";
import { Button } from "../../../../components/Button/Button";
import { Breadcrumb } from "../../../../components/Breadcrumb/Breadcrumb";
import { typesRequestErrors } from "../../../../constants/error.constant";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

window.scrollTo = jest.fn();

let initState = {
  procedures: {
    requestInquiryAll: null,
    proceduresAll: null,
    profile: null,
    errors: {
      [typesRequestErrors.getInformationProcedures]: {
        isError: false,
        msg: null,
      },
    },
  },
  texts: {
    procedures: {
      phoneText: "666",
    },
  },
};

const historyMock = {
  push: jest.fn(),
  location: { state: { status: {error: true} } },
  listen: jest.fn(),
};
const onClose = jest.fn().mockImplementation(() => {});

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <ProceduresContext.Provider value={{ onClose }}>
      <Provider store={store}>
        <BrowserRouter>
          <MainProcedures history={historyMock} />
        </BrowserRouter>
      </Provider>
    </ProceduresContext.Provider>
  );
};

describe("Test <MainProcedures/>", () => {
  beforeEach(() => {
    store = mockStore(initState);
  });

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("should close modal", () => {
    const wrapper = handleMount(initState);

    act(() => {
      wrapper.find(ModalSolicitude).props().onClose();
    });
    wrapper.update();
  });

  

  describe("should close sidebar", () => {
    test("should close on button", () => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Sidebar).props().onClose();
      });
      wrapper.update();
    });
    test("should close on outside of sidebar", () => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Sidebar).props().outside();
      });
      wrapper.update();
    });
  });

  test("should open sidebar", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(".procedures-btns").find(Button).last().props().onClick();
    });
    wrapper.update();
  });

  test("should redirect", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(".procedures-btns").find(Button).first().props().onClick();
    });
    wrapper.update();
  });

  describe("should back to dashboard and current page", () => {
    test("should back to dashboard", () => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Breadcrumb).props().onBack();
      });
      wrapper.update();
    });
    test("should go current page", () => {
      const wrapper = handleMount(initState);
      act(() => {
        wrapper.find(Breadcrumb).props().onItem({ detail: "/" });
      });
      wrapper.update();
      //   expect(historyMock.push.mock.calls[0][0]).toEqual("/");
      act(() => {
        wrapper.find(Breadcrumb).props().onItem({ detail: "/other" });
      });
      wrapper.update();
      //   expect(historyMock.push.mock.calls[0][0]).toEqual("/");
    });
  });
  test("should card redirect", () => {
    initState = {
      ...initState,
      procedures: {
        ...initState.procedures,
        proceduresAll: [
          {
            path: "/procedures-flows/study-record/information",
            icon: "description",
            type: "study-record",
            title: "Constancia de estudio",
            disabled: false,
            isContent: true,
            content: {
              text: "Documento con el que cada estudiante puede acreditar toda su trayectoria académica. En dicho documento se certifica oficialmente los estudios que una persona ha cursado con éxito.",
              price: "$480.00 - $590.00",
            },
          },
          {
            path: "/procedures-flows/certificate-study/information",
            icon: "star",
            type: "certificate-study",
            title: "Certificado de estudios",
            disabled: false,
            isContent: true,
            content: {
              text: "Documento donde se declara oficialmente que haz acreditado tu plan de estudios, este puede ser completo o parcial, incluye fechas, calificaciones, nombre de las unidades de aprendizaje, sellos y firmas de autoridades.",
              price: "$480.00 - $590.00",
            },
          },
        ],
      },
    };
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(CardProcedure).first().props().onClick();
    });
    wrapper.update();
  });

  test("should open modal", () => {
    const wrapper = handleMount(initState);
  });
});
