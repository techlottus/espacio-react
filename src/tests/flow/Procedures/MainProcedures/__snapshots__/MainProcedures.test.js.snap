// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <MainProcedures/> should init component 1`] = `
Array [
  <div
    className="navbar"
  >
    <lottus-navbar />
  </div>,
  <div
    className="main-procedures-container"
  >
    <div
      className="breadcrumb"
    >
      <lottus-breadcrumb />
    </div>
    <div>
      <h2
        className="procedures-title"
      >
        Tr√°mites / Servicios
      </h2>
    </div>
    <div
      className="procedures-btns"
    >
      <div
        className="procedures-btn"
      >
        <lottus-button />
      </div>
      <div
        className="procedures-btn"
      >
        <lottus-button />
      </div>
    </div>
    <div
      className="cards-procedures"
    />
  </div>,
  "",
  <div
    className="footer"
    style={
      Object {
        "bottom": "0",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <lottus-footer />
  </div>,
]
`;
